### PR TITLE
Improve meson build and fixes

### PR DESCRIPTION
**Pull request by NathanBnm**
This PR improves the way your app is build with meson with a better integration.
I made a few fixes too. Let me know if you have any questions before merging.

### Changelog
- Add .editorconfig with elementary default configuration
- Capitalize source folders (convention)
- Improve meson build system integration
- ~~Remove unused translation files (better maintainability)~~
- Improve translation files integration (metadata are now in the `extra` folder)
- Remove deprecated/unused constants in `Constants.vala`

I'll have to make a few translation fixes in another PR once you'll have merged this one so please wait for translation updates before publishing a new release then :smile:

**This PR is ready for review**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,2 @@
-build/*
-debian/build/*
-debian/com.github.arshubham.cipher.debhelper.log
-debian/com.github.arshubham.cipher.substvars
-debian/com.github.arshubham.cipher/
-debian/debhelper-build-stamp
-debian/files
-.buildconfig
-.vscode/
```
Comment (arshubham) [inline .gitignore]: I would like to keep .vscode/ here.

File: data/meson.build [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
+        join_paths('icons', i, meson.project_name() + '.svg'),
+        install_dir: join_paths(get_option('datadir'), 'icons', 'hicolor', i + 'x' + i, '@2', 'apps')
+    )
 endforeach
 
-install_data(
-    join_paths('images', meson.project_name() + '.svg'),
-    install_dir: join_paths(get_option('datadir'), 'Cipher', 'images')
-)
+#install_data(
```
Comment (arshubham) [inline data/meson.build]: Please remove commented code.

**Comment by arshubham** [comment]
I would also like to keeps all language codes in  `po/LINGUAS` . I am thinking of setting up a weblate instance for all my projects.

**Comment by NathanBnm** [comment]
@arshubham I made the changes.

File: po/nl.po [hunk 3]
```diff
@@ -1,198 +1,121 @@
...
-msgstr "vercijferen;kluis;cryptografie;"
-
-#: data/com.github.arshubham.cipher.desktop.in:16
-msgid "About Cipher"
-msgstr "Over Cipher"
-
-#: data/com.github.arshubham.cipher.appdata.xml.in:9
-msgid "Shubham Arora"
-msgstr "Shubham Arora"
+#~ msgid "com.github.arshubham.cipher"
```
Comment (arshubham) [inline po/nl.po]: Why is this commented? Also noticed it in several other translations.

**Comment by arshubham** [comment]
@NathanBnm Is there a way to test translations without changing system language?

File: po/nl.po (hunk 3 above)
Comment (NathanBnm) [reply po/nl.po]: I forgot to remove it after translation update. I'll fix this now.

**Comment by NathanBnm** [comment]
@arshubham no but I already tested with French translation and this is working fine. I'll just have to update some of them in another PR after this one.

**Comment by NathanBnm** [comment]
@arshubham resolved now

**Comment by arshubham** [comment]
Great, just added gettext action will update appdata and check if it is working.

**Comment by arshubham** [comment]
@NathanBnm You can update the translations now.

**Pull request merged**